File: binary-tree-tilt.cpp
//Runtime: 16 ms
class Solution {
public:

    int tilt = 0;

    int solve(TreeNode* root)
    {
        if(!root)
            return 0;

        int l = solve(root->left);
        int r = solve(root->right);
        tilt += abs(l-r);

        return l+r+root->val;
    }

    int findTilt(TreeNode* root) {
        if(!root)
            return 0;
        solve(root);
        return tilt;
    }
};

File: binary-tree-tilt.py
# Runtime: 79 ms
class Solution(object):
    def findTilt(self, root):
        self.res = 0

        def solve(root):
            if not root:
                return 0

            a = solve(root.left)
            b = solve(root.right)
            self.res += abs(a - b)
            return a + b + root.val

        solve(root)
        return self.res
